packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -156,29 +243,50 @@ class CupertinoTabScaffold extends StatefulWidget {
 }
 
 class _CupertinoTabScaffoldState extends State<CupertinoTabScaffold> {
-  int _currentPage;
+  CupertinoTabController _controller;
 
   @override
   void initState() {
     super.initState();
-    _currentPage = widget.tabBar.currentIndex;
+    _updateTabController();
   }
 
+  void _updateTabController() {
+    final CupertinoTabController newController =
+      // User provided a new controller, update [_controller] with it.
+      widget.controller
+      // User didn't provide a new controller, keep using [_controller].
+      ?? _controller
+      // This only happens in [initState]. If there's no existing controller,
+      // create one using [widget.tabBar.currentIndex] as its initial index.
+      ?? CupertinoTabController(selectedIndex: widget.tabBar.currentIndex);
+
+    assert(() {
+        if (newController.currentIndex >= widget.tabBar.items.length) {
+          throw FlutterError(
+            'current index ${newController.currentIndex} is out of bounds.'
```

Comment:
`assert(newController.currentIndex < widget.tabBar.items.length, "The CupertinoTabController's initialIndex ${newController.initialIndex} is out of bounds for the tab bar with ${widget.tabBar.items.length} tabs");`